container_service_extension/configure_cse.py

```diff
@@ -296,20 +321,50 @@ def install_cse(config_file_name, skip_template_creation,
         msg_update_callback.general(msg)
         INSTALL_LOGGER.info(msg)
 
-        # create amqp exchange if it doesn't exist
-        amqp = config['amqp']
-        _create_amqp_exchange(amqp['exchange'], amqp['host'], amqp['port'],
-                              amqp['vhost'], amqp['ssl'], amqp['username'],
-                              amqp['password'],
-                              msg_update_callback=msg_update_callback)
+        # Setup extension message protocol
+        if utils.use_mqtt_protocol(config):
+            mqtt_ext_manager = MQTTExtensionManager(client)
+            description = _construct_cse_extension_description(
+                config['vcd']['api_version'])
+            ext_info = mqtt_ext_manager.setup_extension(
+                ext_name=server_constants.CSE_SERVICE_NAME,
+                ext_version=server_constants.MQTT_EXTENSION_VERSION,
+                ext_vendor=server_constants.MQTT_EXTENSION_VENDOR,
+                description=description)
+            ext_uuid = mqtt_ext_manager.get_extension_uuid(
+                ext_info['ext_urn_id'])
+            _ = mqtt_ext_manager.setup_api_filter(
+                server_constants.MQTT_API_FILTER_PATTERN, ext_uuid)
+        else:
+            # create amqp exchange if it doesn't exist
+            amqp = config['amqp']
+            _create_amqp_exchange(amqp['exchange'], amqp['host'], amqp['port'],
+                                  amqp['vhost'], amqp['ssl'], amqp['username'],
+                                  amqp['password'],
+                                  msg_update_callback=msg_update_callback)
+
+            # register cse as an api extension to vCD
+            _register_cse_as_extension(  # TODO: see how to use MQTT
```

Comment:
Its slightly confusing; So we should:

- Registering CSE as an extension

- Register rights

only if the message-bus is AMQP? in case of MQTT we do not need to do that?
Can we please add some clarity about it in design doc, if thats actually the case?